mailing_list/static/mailing_list/js/app.js

```diff
@@ -1,134 +1,140 @@
-            }
-        },{
-            id: 'everyone',
-            name: {class: 'World Access:', section: 'World Access:'},
-            description: {
-                class: 'anyone can send and reply to this mailing list.',
-                section: 'anyone can send and reply to this mailing list.'
-            }
-        },{
-            id: 'readonly',
-            name: {class: 'Disabled:', section: 'Disabled:'},
-            description: {
-                class: 'this mailing list is disabled.',
-                section: 'this mailing list is disabled.'
-            }
-        }];
-        ml.accessLevelDisplayNameMap = {};
-        ml.accessLevelDescriptionMap = {};
-        for (var i = 0; i < ml.accessLevels.length; i++) {
-            var accessLevel = ml.accessLevels[i];
-            ml.accessLevelDisplayNameMap[accessLevel.id] = accessLevel.name;
-            ml.accessLevelDescriptionMap[accessLevel.id] = accessLevel.description;
+    $http.get(URL_LISTS).success(function(data){
+      ml.isLoading = false;
+      var length = data.length;
+      var course_list_member_count = 0;
+      for (var i = 0; i < length; i++) {
+        var list = data[i];
+        if(list.is_course_list) {
+          ml.courseList.push(list);
+          course_list_member_count = list.members_count;
+        } else if(!list.sis_section_id || list.cs_class_type == 'N'){
+          ml.nonEnrollmentSectionLists.push(list);
+        } else if(list.sis_section_id || list.cs_class_type == 'E'){
+          ml.enrollmentSectionLists.push(list);
         }
+      }
+      ml.loaded = true;
+    });
 
-        $http.get(URL_LISTS).success(function(data){
-            ml.isLoading = false;
-            var length = data.length;
-            for (var i = 0; i < length; i++) {
-                var list = data[i];
-                if (list.is_primary_section) {
-                    ml.primarySectionLists.push(list);
-                } else {
-                    ml.otherSectionLists.push(list);
-                }
-            }
-            ml.loaded = true;
-        });
+    ml.hasCourseEmailList = function(){
+      return ml.courseList.length > 0;
+    };
 
-        ml.hasMultiplePrimarySections = function() {
-          return ml.primarySectionLists.length > 1;
-        };
+    ml.hasMultipleEnrollmentSections = function() {
+      return ml.enrollmentSectionLists.length > 1;
+    };
 
-        ml.hasPrimarySections = function() {
-            return ml.primarySectionLists.length > 0;
-        };
+    ml.hasNonEnrollmentSections = function() {
+      return ml.nonEnrollmentSectionLists.length > 0;
+    };
 
-        ml.hasOtherSections = function() {
-            return ml.otherSectionLists.length > 0;
-        };
+    ml.isSisList = function(list){
+      return (list.cs_class_type == 'N');
+    };
```

Comment:
Looks like we are checking for type 'N' here. The name is a bit confusing as it would appear we are checking for sis_id - can we rename it ?